src/alspgrad.jl

```diff
@@ -47,7 +47,7 @@ immutable ALSGradUpdH_State{T}
     WtX::Matrix{T}    # W'X  (pre-computed)
     WtWD::Matrix{T}   # W'W * D
 
-    function ALSGradUpdH_State(X, W, H)
+    function ALSGradUpdH_State{T}(X, W, H) where T <: Real
```

Comment:
The 0.5-compatible syntax hasn't been deprecated in 0.6, so this can be `ALSGradUpdH_State{T<:Real}(X, W, H)`. Also the `new(...)` needs to become `new{T}(...)`.